### PR TITLE
Enable eslint-plugin-react with recommended settings & forbid button

### PR DESCRIPTION
**Pull request by luchillo17**
Signed-off-by: Carlos Esteban Lopez <lcarlosesteb@vmware.com>

## Hey, I just made a Pull Request!

Enable the `eslint-plugin-react` for non-test files and forbid `<button>` tag in favor of MUI's `<Button>`.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by freben** [comment]
Hi! An early remark: I do not think that we can do this in the CLI itself. The CLI is shipped to all customers, and whatever rules we put in here will be instantly forced upon the world and probably lead to widespread breakage. The value provided vs the pain inflicted is probably not worth it.

You may be able to add this to the root `.eslintrc.js`, but bear in mind that a given rule typically isn't merged "across layers", so if the CLI provides a rule and the root has the same rule, there might be interactions there that you'll have to think about.

**Comment by luchillo17** [comment]
I would expect the rules in CLI would override the root ones, right? I went for CLI because I wanted to apply these rules only to the UI role packages, so better do it in the root so only applies to this repo?

**Comment by freben** [comment]
Again, it's hard at this point to change lint rules _in the cli_ specifically. For the main repo itself, I'm happy to more and more move toward the most commonly well known and trusted presets.

**Comment by freben** [comment]
Adding rules to individual packages can be done like this:

https://github.com/backstage/backstage/blob/b381f3fd48f0c9ded1e99eb376ab9678d8801e1e/packages/codemods/.eslintrc.js#L1

**Comment by luchillo17** [comment]
The CLI way was easier because there was already a switch case for the `package.json` role property for each package, I would prefer to just use it in the root instead of going into each UI package/plugin to add and manage rules, it's not like the rules pertaining to React would interfere with backend code right?

**Comment by freben** [comment]
We already have frontend related ones in the root (testing-library). Should be ok

File: packages/core-components/src/layout/Sidebar/Items.test.tsx [hunk 1]
```diff
@@ -71,7 +71,7 @@ describe('Items', () => {
     it('should render a button with custom style', async () => {
       expect(
         await screen.findByRole('button', { name: /create/i }),
-      ).toHaveStyle(`background-color: ${hexToRgb('2b2a2a')}`);
+      ).toHaveStyle(`background-color: transparent`);
```
Comment (luchillo17) [inline packages/core-components/src/layout/Sidebar/Items.test.tsx:74]: This test was wrong from the start, `hexToRgb` expects a hex number with the symbol, so it should have been `hexToRgb('#2b2a2a')`, now with Button it should be transparent and let the container define the color.

Comment (luchillo17) [reply packages/core-components/src/layout/Sidebar/Items.test.tsx:74]: Visually it looks the same to me.

**Comment by luchillo17** [comment]
@tudi2d Checks seem to have stopped working, can you see if you can trigger a run?

**Comment by freben** [comment]
Github seem to be having problems with actions never triggering...

File: .eslintrc.js [hunk 2]
```diff
@@ -18,7 +18,7 @@ var path = require('path');
 
 module.exports = {
   root: true,
-  plugins: ['notice', 'testing-library'],
+  plugins: ['@spotify', 'notice', 'react', 'testing-library'],
```
[the remark this answers is not in the document]
Comment (luchillo17) [reply .eslintrc.js:21]: At least doesn't trigger any error so the project is already compliant with it

[the remark this answers is not in the document]
Comment (luchillo17) [reply .eslintrc.js:21]: I was looking at this MD https://github.com/spotify/web-scripts/blob/master/packages/eslint-config/README.md

[the remark this answers is not in the document]
Comment (luchillo17) [reply .eslintrc.js:21]: The one you mentioned only has a single rule, I was going for `@spotify/eslint-config`, should I remove it?

**Comment by luchillo17** [comment]
> We already have frontend related ones in the root (testing-library). Should be ok

Maybe it does, seems like the rule runs on the tests, was trying one for `<p>` and gave me a warning on `core-app-api` in a test, if I were to fix it I would have to install MUI on that project and given it's for a test it makes no sense to install it, do you know how to limit the rule to non-test files?

File: .eslintrc.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (freben) [reply .eslintrc.js:21]: Aha, OK let's keep it

**Review by freben — APPROVED** [review]

**Pull request merged**